web/blog/tenzir-v4.6/index.md

```diff
@@ -0,0 +1,235 @@
+date: 2023-12-01
+tags: [release, context, enrich, syslog, tcp, parse, python]
+comments: true
+---
+
+[Tenzir v4.6](https://github.com/tenzir/tenzir/releases/tag/v4.6.0) is here, and
+it is our biggest release yet. The headlining feature is the all-new **context**
+feature, powered by the `context` and `enrich` operators and the new **context
+plugin** type.
+
+![Tenzir v4.6](tenzir-v4.6.excalidraw.svg)
+
+<!-- truncate -->
+
+## Enrich with Contexts
+
+Contexts enable enriching events in real-time with additional information.
+
+This is best explained on an example. We'll use the publicly available [Feodo IP
+Block List](https://feodotracker.abuse.ch) to enrich events with information
+that have a source IP address that is on the block list.
+
+```text {0} title="Create a 'lookup-table' context named 'feodo'"
+context create feodo lookup-table
+```
+
+```text {0} title="Fill the 'feodo' context with information"
+from https://feodotracker.abuse.ch/downloads/ipblocklist_aggressive.csv read csv --allow-comments
+| context update feodo --key dst_ip
+```
+
+```text {0} title="Enrich Suricata events with the 'feodo' context"
+export
+| where #schema == /suricata.*/
+| enrich feodo --field dest_ip
+```
+
+```json {0} title="Possible output"
+{
+  "timestamp": "2023-11-30T14:52:57.716360+0200",
+  "event_type": "alert",
+  "dest_ip": "167.179.103.206",
+  // ...
+  "feodo": {
+    "key": "167.179.103.206",
+    "context": {
+      "first_seen_utc": "2023-11-03T07:55:23",
+      "dst_ip": "167.179.103.206",
+      "dst_port": 2083,
+      "c2_status": "offline",
+      "last_online": "2023-11-03T00:00:00",
+      "malware": "Pikabot"
+    },
+    "timestamp": "2023-11-30T14:58:23.172832+0200"
+  }
+}
+```
+
+Contexts are—just like formats, connectors, and operators—designed to be a
+building block of TQL. We're starting out with just the open-source
+`lookup-table` context plugin, and already have plans for more context plugins
+and other operators besides `enrich` that use context plugins in the near
+future. Stay tuned!
+
+:::info Follow our Blog Post Series
+We're so excited about enrichments that we wrote an entire blog post series on
+it. The [first post](/blog/enrichment-complexity-in-the-wild) is already live,
+and sets the scene. The second post will soon follow to explain in-depth just
+how Tenzir makes enrichments easy. Stay tuned!
+:::
+
+## Onboard Data Faster Than Ever
+
+Want to read CEF over Syslog from TCP? Not a problem with four all-new features
+of Tenzir v4.6—are you able to spot all four?
+
+FIXME: Why is `content --into content` not the default?
+
+```
+from tcp://localhost:514 read syslog
+| parse content --into content cef
+```
+
+Here's an example input and output:
+
+```syslog {0} title="Input"
+Nov 13 16:59:59 host123 FOO: CEF:0|FORCEPOINT|Firewall|6.6.1|0|Generic|0|deviceExternalId=Master FW node 1 dvc=10.1.1.40 dvchost=10.1.1.40 msg=log server connection established deviceFacility=Logging System rt=Jan 17 2020 08:52:10
```

Comment:
I made a note about this here: https://github.com/tenzir/tenzir/pull/3665#discussion_r1410669284

Reply:
I discussed this briefly with Jannis in a call just now, and we both agree that this is a wontfix for the current release and that we will fix it for a later one.